modules/RTC/TPCUtils.js

```diff
@@ -34,19 +35,93 @@ export class TPCUtils {
+                    scalalabilityModeSettings?.useL3T3
+                        && (this.codecSettings[codec].useL3T3 = scalalabilityModeSettings.useL3T3);
+                } else {
+                    this.codecSettings[codec].scalabilityModeEnabled = false;
+                }
+            }
+        }
+    }
+
+    /**
+     * Calculates the configuration of the first encoding when the client is configured to do full SVC for VP9 and AV1
+     * codecs depending on the resolution that is being requested.
+     *
+     * @param {JitsiLocalTrack} localVideoTrack - The local video track.
+     * @param {number} newHeight - The resolution that needs to be configured for the local video track.
+     * @param {CodecMimeType} codec - The video codec.
+     * @returns {Object} configuration.
+     */
+    _calculateActiveEncodingParamsForSvc(localVideoTrack, newHeight, codec) {
+        const codecBitrates = this.codecSettings[codec].maxBitratesVideo;
+        const localTrack = localVideoTrack.getTrack();
+        const { height } = localTrack.getSettings();
+        const desktopShareBitrate = this.pc.options?.videoQuality?.desktopBitrate || DESKTOP_SHARE_RATE;
+        const isScreenshare = localVideoTrack.getVideoType() === VideoType.DESKTOP;
+        const scalabilityMode = isScreenshare && this._isScreenshareBitrateCapped(localVideoTrack)
+
+            // The endpoint needs to send only 1 spatial layer with temporal scalability for low fps SS.
+            ? VideoEncoderScalabilityMode.L1T3
+
+            // For camera and high fps screenshare, we need both spatial and temporal scalability.
+            : this.codecSettings[codec].useL3T3
+                ? VideoEncoderScalabilityMode.L3T3 : VideoEncoderScalabilityMode.L3T3_KEY;
+        const maxBitrate = isScreenshare
+            ? this._isScreenshareBitrateCapped(localVideoTrack) ? desktopShareBitrate : codecBitrates.ssHigh
+            : codecBitrates.high;
+
+        const config = {
+            active: newHeight > 0,
+            maxBitrate,
+            scalabilityMode,
+            scaleResolutionDownBy: HD_SCALE_FACTOR
         };
 
-        // Check if the max. bitrates for video are specified through config.js videoQuality settings.
-        // Right now only VP8 bitrates are configured on the simulcast encodings, VP9 bitrates have to be
-        // configured on the SDP using b:AS line.
-        this.videoBitrates = bitrateSettings ?? standardBitrates;
-        this.encodingBitrates = this.videoBitrates.VP8 ?? this.videoBitrates;
+        if (newHeight >= height || newHeight === 0 || isScreenshare) {
+            return config;
+        }
+
+        if (newHeight >= height / SD_SCALE_FACTOR) {
+            config.maxBitrate = codecBitrates.standard;
+            config.scalabilityMode = this.codecSettings[codec].useL3T3
+                ? VideoEncoderScalabilityMode.L2T3 : VideoEncoderScalabilityMode.L2T3_KEY;
```

Comment:
Sorry I'm a bit out of my depth here, but why are we setting the mode to L2T3 or L2T3_KEY if useL**3**T3 is set?

Reply:
I am using useL3T3 in a broader sense, i.e., if L3T3 (and not L3T3_KEY) is used when full resolution is requested. If the highest layer needs to be suspended then the corresponding scalabilityMode for sending the SD stream would be L2T3 and not L2T3_KEY. I changed the config variable to `useL3T3Mode` if that helps.

Reply:
Sorry, that didn't help :-) Since this translates into a setting that users can set in config.js we need a better name IMHO.

I cannot provide one because it's not 100% clear to me what it really does...

Reply:
This setting is experimental until we figure out which mode is better for full SVC, `L3T3` or `L3T3_KEY`. For anyone actually interested in understanding the differences, I have the referenced the doc that explains the modes here https://github.com/jitsi/lib-jitsi-meet/pull/2350/files#diff-66113b9503c3b54c96306116bcd0c8e3bace7c11861d47f7f91887e0612ae491R3. I honestly can't think of a better name myself.

Reply:
I may be missing something here, but maybe the config name should be about whether or not the K-SVC is used? for example `use_K_SVC` ?

Then:
```
config.scalabilityMode = this.codecSettings[codec].use_K_SVC
                ? VideoEncoderScalabilityMode.L2T3_KEY : VideoEncoderScalabilityMode.L2T3;
```

It seems in various places that the config is used to decide whether K-SVC is to be used or not, but currently it's meaning is the inversion of that.

Reply:
In case having a default for the config is a problem, you can check for `undefined` to tell if the default of `true` in this case should be used.

Reply:
We want to make K-SVC the default mode to be used unless its overridden through config.js. `use_K_SVC` makes sense and users will have to switch it to false in the config to try the other mode.

Reply:
Done, renamed the setting and inverted the defaults.